src/v/kafka/server/materialized_partition.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+#include "kafka/server/partition.h"
+#include "storage/log.h"
+
+namespace kafka {
+class materialized_partition final : public kafka::partition::impl {
+public:
+    explicit materialized_partition(storage::log log)
+      : _log(log) {}
+
+    const model::ntp& ntp() const final { return _log.config().ntp(); }
+    model::offset start_offset() const final {
+        return _log.offsets().start_offset;
+    }
+
+    model::offset high_watermark() const final {
+        return _log.offsets().dirty_offset;
+    }
+
+    model::offset last_stable_offset() const final {
+        return _log.offsets().dirty_offset;
```

Comment:
Is this right?

Reply:
yes, we can read up to dirty offset, in materialized partitions.